OpenEdX.xcodeproj/project.pbxproj

```diff
@@ -375,6 +373,24 @@
 			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n# This output is used by Xcode 'outputs' to avoid re-running this script phase.\necho \"SUCCESS\" > \"${SCRIPT_OUTPUT_FILE_0}\"\n";
 			showEnvVarsInLog = 0;
 		};
+		DB97C0542B002EF00035C36F /* Process Config */ = {
```

Comment:
@mumer92 Now to switch between Stg and Prod version I need to edit config.yml file manually, right? What do think about adding parameter to process_config.py and call it with own config.yml file (every file will point out to particular folder prod/stage/prod_test for example in existing 'edx-mobile-config' repo) for every Scheme? This way you just select another Scheme in Xcode to switch between environments.

Reply:
<img width="795" alt="Screenshot 2023-11-13 at 17 16 04" src="https://github.com/openedx/openedx-app-ios/assets/37253/6adc1210-a2c3-4ad8-9268-b00917908b08">
This is how we can modify 'Process Config' Build Phase to check current Scheme. 
If user want to modify config he could point any Scheme to own config files path.
It looks similar how to determine 'path' in config.yaml you suggested but it could be done only once and don't need to change 'environment' in config.yaml every time you would jump from prod to stg.
And this way saves Schemes which are critical for openEdx how @volodymyr-chekyrta said.

Reply:
Instead "Unknown configuration" we can call script with some 'default' path

Reply:
i have removed `process_config.sh` in favour of directly running script from build phase, passing `$CONFIGURATION` as argument to script and also updated `process_config.py` to be able to handle `$CONFIGURATION`. if configuration is not provided, it runs with the config provided inside `config.yaml`

Reply:
@mumer92 in this case if user wants to add new Scheme he needs to edit script file, right? Also script strongly determines folders name as 'prod', 'prod-test' and 'stage' (what if user wants to have 'sandbox'?). If we could create new Scheme and define which folder is attached to it in one place (in Xcode build phase) it's better from my view. What are your thoughts?
Also one question is what is pros to have this 'config.yaml' file - why 'directory' can't be a part of 'path' script's  parameter in build phase too?
Thank you

Reply:
We generally had 3 configs in edX project, but you are welcome to add them to project. The idea is to have to do minimal to no change, in that way, you will just chose scheme and config is applied for you, based on scheme.

in `config.yaml` the mappings are provided as follows
```
config_mapping: 
  dev: 'dev'
  prod: 'prod'
  stage: 'stage'
```

the idea for providing the mappings it to have a flexibility, e.g.
```
config_mapping: 
  dev: 'prod_test'
  prod: 'prod'
  stage: 'stage'
```

if someone has `prod_test` instead of `dev`, they would be able to specify the structure here and mappings are read from the script and script looks for the files based on the value, 
e.g. if you have 3 folders, `prod - prod_test - stage`, you will provide these names as mappings here and script then maps the directory name to the appropriate key, in this case, if you run Xcode with `DebugDev` scheme, script will read file mappings and treat them as directory name and find yaml files inside that directory and loads them in script.

for reference, please see following:
https://github.com/openedx/openedx-app-ios/blob/7b09d047af8827a1f6cd7a9e9e440fb2464c94dc/process_config.py#L214C1-L222C20

Reply:
Hello @mumer92 
Flexability is exactly what I'm saying about. When I want to add one more Scheme 'Sandbox' I need to to config.yaml and add mapping there (one more question what exists still is what if this 'sandbox' config doesn't live in defined there 'config_directory'?). After that I need to edit python script by adding configuration name and mapping value.
Instead of all this stuff I'm showing how we can do it without config.yaml and python script editing
<img width="774" alt="Screenshot 2023-11-15 at 11 16 33" src="https://github.com/openedx/openedx-app-ios/assets/37253/6ef968b5-3057-4992-a700-3e5114b80af8">
If we need to add new Scheme (or point a Scheme out to other config) you just add one line here:
`"DebugSandbox" | "RelesaseSandbox" ) python process_config.py "$PATH_TO_CONFIG/sandbox" ;;` (and here PATH_TO_CONFIG can be any - default or user's)
So my question is why python script should construct path to config from config.yaml and from mapping if it can get path as parameter and work with it?
@volodymyr-chekyrta please share your thoughts here

Reply:
@rnr I dont think much of developer effort would be required in this case,
e.g. there are 3 configs provided as default on project, and if developer wants to add another config, they can either modify this function to handle configuration and provide mappings, or break down the case:

```
def get_configuration_name(configuration, config_mapping):
    if configuration in ["DebugDev", "ReleaseDev"]:
        return config_mapping.get("dev")
    elif configuration in ["DebugStage", "ReleaseStage"]:
        return config_mapping.get("stage")
    elif configuration in ["DebugProd", "ReleaseProd"]:
        return config_mapping.get("prod")
    else:
        return None
```

or e.g. they can update the `config.yaml` to update the mapping, and i dont think either apporach would take much of developer effort, could be done in a minute or 2.

Reply:
@mumer92 
Why did we go from discussing flexibility and usability to developer effort? Let's please get back to discussing the advantages and disadvantages of approaches.
I don't like the fact that changing/adding a schema in Xcode causes the need to go and fix a python script - in my opinion it's not about flexibility. Why we create a rigid connection between scheme names and python script?
My second question is why do we need to create an additional config.yaml file if we already have a build phase dedicated to configuration, which can contain the schema-config mapping? What is pros of this file?
Thank you

Reply:
I think multiple cases are being discussed here and are getting confused with each other.

@rnr How about if we can finalize this PR first and then discuss it in more detail over a call to make it more mature? I'll be looking into it with @mumer92. Scenario, we have a large community that has already published their apps with their configurations (with a specific / existing format). We want to have something where there will be minimal change while shifting to the new codebase. 

We don't have the luxury of freedom here to follow anything. So given the circumstances, we have to come up with the best possible solution, which might not be the best solution.

cc @miankhalid @touchapp